src/gpu/generic/sycl/ref_reorder.hpp

```diff
@@ -98,14 +98,22 @@ struct ref_reorder_t : public gpu::generic::sycl::primitive_t {
         }
 
         bool scales_ok() const {
-            const std::vector<int> supported_args
-                    = {DNNL_ARG_SRC, DNNL_ARG_DST};
+            if (!attr()->scales_.has_default_values(DNNL_ARG_FROM)) {
+                // Check if scales are applied and then check the data type
+                // otherwise either data type can come up as data_type::undef
+                // returning false unecessarily
+                const auto src_scale_dt = attr()->scales_.get_data_type(
+                        DNNL_ARG_ATTR_SCALES | DNNL_ARG_FROM);
```

Comment:
I still insist current line return `data_type::undef` unconditionally and passes it into `is_supported_type`. Depending on the handling of `undef`, it either always `false` or `true`. Which one is it?

```suggestion
                        DNNL_ARG_FROM);
```